pkg/resources/user_parameters.go

```diff
@@ -77,11 +77,12 @@ var (
 )
 
 type parameterDef[T ~string] struct {
```

Comment:
It's not a part of this PR, but we should move parameterDef to a common file. It's also used in schemas and databases.

Reply:
We already have SNOW-1645342

Reply:
Okay, I got confused because TODOs refer only to user parameter variables.